src/UrlParser.elm

```diff
@@ -268,8 +268,16 @@ oneOfHelp choices chunks formatter =
         Err _ ->
           oneOfHelp otherParsers chunks formatter
 
-        Ok answerPair ->
-          Ok answerPair
+        Ok (chunks', result) ->
+          case chunks'.rest of
+            [] ->
+              Ok (chunks', result)
+
+            [""] ->
+              Ok (chunks', result)
+
+            _ ->
+              oneOfHelp otherParsers chunks formatter
```

Comment:
I could be wrong, but shouldn't this line return Err if there's more pieces of the URL left over after a given choice succeeds?  Similar to in [parse](https://github.com/capicue/url-parser/blob/cb99ba1dddd8c777ef3f67e7b82c22dae39e5fc1/src/UrlParser.elm#L93).

```elm
Err <|
    "The parser worked, but /"
        String.join "/" _ ++ " was left over."
```